### PR TITLE
HPCC-26666 NodeJS version 16

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dockerfiles/buildall-common.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # Build script to create and publish Docker containers corresponding to a GitHub tag
 # This script is normally invoked via GitHub actions, whenever a new tag is pushed
 
-BASE_VER=7.12                                   # The docker hub label for the platform-build-base image. Changes rarely.
+BASE_VER=8.6                                    # The docker hub label for the platform-build-base image. Changes rarely.
```
[the remark this answers is not in the document]
Comment (richardkchapman) [reply dockerfiles/buildall-common.sh:23]: It's somewhat unrelated, but it reflects the first platform version that uses this base version.

[the remark this answers is not in the document]
Comment (richardkchapman) [reply dockerfiles/buildall-common.sh:23]: (and note that this is targetting master, hence the next 8.4.x minor release will NOT be using this)

[the remark this answers is not in the document]
Comment (jakesmith) [reply dockerfiles/buildall-common.sh:23]: ok, fair enough.

**Review by jakesmith — APPROVED** [review]
@richardkchapman - looks good.

File: dockerfiles/platform-build-base/Dockerfile [hunk 2]
```diff
@@ -83,7 +83,7 @@ RUN apt-get install -y \
   jq
 
 RUN curl https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
-RUN apt-add-repository "deb https://deb.nodesource.com/node_12.x $(lsb_release -sc) main"
+RUN apt-add-repository "deb https://deb.nodesource.com/node_14.x $(lsb_release -sc) main"
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply dockerfiles/platform-build-base/Dockerfile:86]: (We have github actions testing builds against v12, 14 and 16 already)

**Comment by richardkchapman** [comment]
Repushed to target v16.

This is going to need updating after https://github.com/hpcc-systems/HPCC-Platform/pull/15453 is merged/upmerged

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**